docs/source/installation.rst

```diff
@@ -2,152 +2,125 @@ Installation
+        2. Activate the new virtual environment (default named `ivadomed_env`)
 
-::
+        ::
 
-    conda activate IvadoMedEnv
+            source ivadomed_env/bin/activate
 
-Step 4 : Install from source
-++++++++++++++++++++++++++++++++++++++
 
-::
+    .. tab:: Compute Canada HPC
 
-    pip install -e .
+        There are numerous constraints and limited package availabilities with ComputeCanada cluster environment.
 
-Approach 2: Venv
------------------
+        It is best to attempt ``venv`` based installations and follow up with ComputeCanada technicall support as MANY specially compiled packages (e.g. numpy) are exclusively available for Compute Canada HPC environment.
 
-Step 1: Setup Python Virtual Environment.
-+++++++++++++++++++++++++++++++++++++++++
+        If you are using `Compute Canada <https://www.computecanada.ca/>`_, you can load modules as `mentioned here <https://intranet.neuro.polymtl.ca/computing-resources/compute-canada#modules>`_ and `also here <https://docs.computecanada.ca/wiki/Utiliser_des_modules/en#Loading_modules_automatically>`_ .
 
-``ivadomed`` requires Python >= 3.6 and <3.9 (If you are using `Compute Canada <https://www.computecanada.ca/>`_, you can load modules (e.g. python 3.9) as mentioned  `here <https://intranet.neuro.polymtl.ca/computing-resources/compute-canada#modules>`_ and `also here <https://docs.computecanada.ca/wiki/Utiliser_des_modules/en#Loading_modules_automatically>`_ ). We recommend
-working under a virtual environment, which could be set as follows:
 
-::
+Step 2: Install ``ivadomed``
+++++++++++++++++++++++++++++
 
-    virtualenv venv-ivadomed
-    source venv-ivadomed/bin/activate
 
+.. tabs::
 
-.. warning::
-   If the default Python version installed in your system does not fit the version requirements, you might need to specify a version of Python associated with your virtual environment:
+    .. tab:: Pypi Installation
 
-   ::
+        Install ``ivadomed`` and its requirements from
+        `PyPI <https://pypi.org/project/ivadomed/>`__:
 
-     virtualenv venv-ivadomed --python=python3.6
+        ::
 
-Step 2: Clone the `ivadomed <https://github.com/ivadomed/ivadomed>`_ repository.
-+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+            pip install --upgrade pip
 
+            pip install ivadomed
 
-::
+    .. tab:: Repo Installation (Advanced or Developer)
 
-    git clone https://github.com/ivadomed/ivadomed.git
-    cd ivadomed
- 
+        Bleeding-edge developments are available on the project's master branch
+        on Github. Installation ``ivadomed`` from source:
 
-Step 3: Install PyTorch 1.5 and TorchVision
-+++++++++++++++++++++++++++++++++++++++++++
+        ::
 
-If you have a compatible NVIDIA GPU that supports CUDA, run the following command:
+            git clone https://github.com/ivadomed/ivadomed.git
 
-::
-   
-   pip install -r requirements_gpu.txt
+            cd ivadomed
 
+            pip install -e .
```

Comment:
I got this error when installing in `venv`:
`ERROR: packaging 21.2 has requirement pyparsing<3,>=2.0.2, but you'll have pyparsing 3.0.5 which is incompatible.`

This prevent `ivadomed` from running. When running CLI I get:
```
Traceback (most recent call last):
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (pyparsing 3.0.5 (/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages), Requirement.parse('pyparsing<3,>=2.0.2'), {'packaging'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mhbourget/ivadomed_env_969_2/bin/ivadomed", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3252, in <module>
    def _initialize_master_working_set():
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3235, in _call_aside
    f(*args, **kwargs)
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3264, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 585, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 598, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (pyparsing 3.0.5 (/home/mhbourget/ivadomed_env_969_2/lib/python3.8/site-packages), Requirement.parse('pyparsing<3,>=2.0.2'), {'packaging'})
```

Reply:
This issue is weirdly related to venv/pip common dependency version which several fixes are available (simplest is to update pip to 21). Could you check if you have Pip v20? In depth details as well as other many useful solutions are reported by @kousu here: https://github.com/neuropoly/computers/issues/218.

We will probably update the dependencies to be more specific to globally address this odd problem.

Reply:
You are right, I had `pip 20.0.2`.
I did `pip install --upgrade pip` and I now have `pip 21.3.1` and the installation and CLI run without error.

Reply:
Thanks to Nick, we have also added that pyparsing as a requirement and merged to master to include additional details. See #987